### PR TITLE
Create some tests for CodeBox

### PR DESCRIPTION
**Pull request by maexled**
Create some tests for the CodeBox.

~WIP: Currently I want to get to modal when I want to test to submit a code word as bug. Currently I can not access the modal in the tests. Need some help. If we can not solve this, we have to live without a test for submitting a bug.~

### CODE REVIEW

**Comment by delvh** [comment]
https://stackoverflow.com/questions/56227464/how-do-you-open-a-bootstrap-vue-modal-with-vue-test-utils ?

**Comment by maexled** [comment]
> https://stackoverflow.com/questions/56227464/how-do-you-open-a-bootstrap-vue-modal-with-vue-test-utils ?

This solution did not work. I did in in my newest commit. The modal was in something named like a teleport (https://test-utils.vuejs.org/guide/advanced/teleport.html#interacting-with-the-teleported-component). I had to iterate over components, I could not iterate over ids like `'#edit-modal'`. Was a bit tricky.

**Comment by maexled** [comment]
I created some more tests.I think now it is ready for review.

**Review by Gr33ndev — APPROVED** [review]

File: tests/unit/CodeBox.spec.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    const lines = wrapper.findAll('.code-line');
+    expect(lines.length).toBe(newCode.words.filter((word) => word.word == WordType.NEWLINE).length + 1);
+
+    const words = wrapper.findAll('.code-word');
+    expect(words.length).toBe(newCode.words.filter((word) => word.word != WordType.TAB && word.word != WordType.NEWLINE).length);
+
+    const spaces = wrapper.findAll('.code-space');
+    expect(spaces.length).toBe(newCode.words.filter((word) => word.word == WordType.SPACE).length);
+  });
+  test('CodeBox shows no feedback when there is none feedback provided', () => {
```
Comment (delvh) [inline tests/unit/CodeBox.spec.ts:93]:
```suggestion
  test('CodeBox shows no feedback when there is no feedback provided', () => {
```

File: tests/unit/CodeBox.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    const selectedWord = 2;
+    const targetWordButton = wrapper.find('#word-' + selectedWord);
+    expect(targetWordButton.exists()).toBe(true);
+
+    targetWordButton.trigger('click');
+    await wrapper.vm.$nextTick();
+
+    const modal = wrapper.getComponent(BModal);
+    const buttons = modal.findAllComponents(BButton);
+    const submitButton = buttons.find((button) => button.text() == "Ok");
```
Comment (delvh) [inline tests/unit/CodeBox.spec.ts:142]: I recommend a less error prone such as a specific style class, id, or attribute. But do not compare by text.
That is the equivalent to testing for a Go error that does not define an interface.

Comment (maexled) [reply tests/unit/CodeBox.spec.ts:142]:
```suggestion
    const submitButton = buttons.find((button) => button.attributes().class.includes("btn-primary"));
```
I can search by btn-primary class which is the Ok button. I do not see more possibilities for this.

Comment (delvh) [reply tests/unit/CodeBox.spec.ts:142]: Far better than what you have now.

**Comment by delvh** [comment]
Do you perhaps want to fix the two Sonarqube suggestions as well here?
(https://sonar.sqa-kube-cluster4.cloudns.nz/component_measures?id=bugfinder&metric=new_technical_debt&selected=bugfinder%3Asrc%2Fcomponents%2FCodeBox.vue&view=list)

**Comment by maexled** [comment]
> Do you perhaps want to fix the two Sonarqube suggestions as well here? (https://sonar.sqa-kube-cluster4.cloudns.nz/component_measures?id=bugfinder&metric=new_technical_debt&selected=bugfinder%3Asrc%2Fcomponents%2FCodeBox.vue&view=list)

done

**Review by delvh — APPROVED** [review]

**Pull request merged**